enhancements/helm3/console.md

```diff
@@ -0,0 +1,144 @@
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: HelmRepository
+metadata:
+  name: cluster
+spec:
+  url: http://my.chart-repo.org/stable
+
+  # optional and only needed for UI purposes
+  displayName: myChartRepo
+
+  # optional and only needed for UI purposes
+  description: my private chart repo
+
+  # set to true if need to be disabled temporarly
+  disabled: true
+```
+
+The console operator would watch for changes on it and reconfigure the chart repository proxy. The console backend already implements a few helm endpoints (including the chart proxy), but the future might require to extract them into a separate service (e.g. to make openshift more modular). Hence, it would be good to define the configuration as the top-level resource detaching API from the implementation. Cluster admin would easily discover the new resource and manage it either via CLI or through UI. 
+
+![Helm Cluster Configuration](assets/openshift-administration-cluster-settings.png)
+
+#### Alternatives
+
+#### 1. The configuration could be embedded into cluster-wide [`Console` config](https://github.com/openshift/api/blob/master/config/v1/types_console.go#L26)
+
+Admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs. It becomes closely coupled with the console. Extracting Helm endpoints into a separate service would require moving the config as well.
+
+#### 2. The configuration could be embedded into [`Console` operator config](https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L26)
+
+Conceptually, the Helm repository URL isn't really an operator configuration, hence this doesn't feel like the right place.
+This approach would have similar issues with the previous alternative - admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs.
+
+#### 3. OLM operator for Helm Configuration. 
+
+Note, the helm charts' repository configuration today exists as a console configuration, which enables Console to proxy to the Helm chart repository URL. Moving it out of Console is outside the scope of this section. 
+
+   * The default helm chart repository URL remains unchanged in the Console configuration.
+   * Admin installs an OLM operator which only provides a `HelmConfig` cluster-scoped CRD
+   * Admin creates a cluster-scoped CR. Note, this isn't very intuitive for the Admin.
+   * Console-operator watches the new `HelmConfig` CR and reconciles.
+   
+Reflections on this approach:
+* We get to avoid changes to `openshift/api`.
+* However, Console operator would have to watch the `HelmConfig` CRD which it doesn't own.
+* Creation of the cluster-scoped `HelmConfig` CR may not be very intuitive for the admin unless we show it in the Cluster Configuration UI in Console.
+* Ideally, the operator should have been pre-installed in the cluster, but that isn't supported.
+
+## How would the UI install charts
```

Comment:
as i read more of this enhancement, i am wondering why we feel obliged to couple our implementation to the console operator at all.  rather than continue to couple things that do not need to be coupled, can we at least specify how it would look when uncoupled?

Reply:
+1

Reply:
They are ~coupled~ related at the moment because we are bringing the Helm experience to Console. 

The `helm` CLI way of listing repos is different than the Console way of listing repos:

* The `helm` CLI looks into the file system for figuring out the available helm chart repositories. ( We'll update the enhancement how the CLI could eventually make use of in-cluster representations ie CRs  )
* The UI would have to look at a REST API to figure that out.

Console Operator reconciling the information isn't coupling per se, it's more of Console being aware of the presence of a new Helm repository.

At the moment, the chart repo url is just an env var which the console proxy uses. We needed this env var because we didn't have a CR to read that information from. 

Once we decide on the `HelmChartRepository` CR, I'll consider using the UI equivalent of `oc get helmchartrepositories` to have the UI rendered.